### ISSUE TITLE
Pressing the button (SW1) drops SRAM voltage too low [FIXED!]

### ISSUE THREAD

**Issue by MouseBiteLabs** [MouseBiteLabs/Game-Boy-MBC3-Multicart#2]
There's an error in the BOM for the MBC3 and MBC5 multicarts (before February 24, 2024). If you built it with R5 and R6 as 10kΩ resistors and C8 as a 0.1uF capacitor, then when you press the button (SW1) while the Game Boy is powered OFF, the voltage on the SRAM supply will drop to 1/2 of the battery voltage as long as you have the button pressed (~1.5V). Thus, the following issues *may* occur if you press the multicart button to change games while the Game Boy is powered OFF only:

- Cartridge may randomly pick one of the games at start-up (last played game is not preserved)
- **Save data may be lost if battery voltage is low enough (but higher than expected for loss of data)**

This is more likely to happen if you are using old, harvested SRAM as compared to the new Alliance parts I specify in the BOM. The reason for this is because the Alliance SRAM's data retention voltage is specified as 1.5V, whereas some of the older SRAM chips that I've checked have this voltage as 2V.

These issues are currently theoretical, and I can't actually get them to happen myself, unless I presumably use a low voltage save battery or specific models of older SRAM. The practical implication of this error is that your save data might be lost earlier than expected, only if you change games when the Game Boy is turned off. If you have the Game Boy on and press the button, the issues will not occur. Luckily, changing the parts is easy, as they're pretty easily accessible. You can do one of the following:

**Fix No. 1 (preferred):**
- Change R5 and R6 to 130kΩ: https://mou.sr/3MjXliy
- Change C8 to 0.01uF: https://mou.sr/3AsRwK1
The 130kΩ resistor is the same one used for R9. You may have extras.
The cap is the same used on the MGBC/DMGC projects, if you've done those you likely have some extras. If you haven't, pretty much any 0.01uF will work fine, provided they fit on the pads.

*Technical explanation: Instead of the SRAM voltage dropping to 1/2 of the battery voltage when the button is pressed, it will drop to 93% of the battery voltage, due to the increased ratio between R1 and R5. This gives plenty of headroom for the SRAM to retain data, even older models. Furthermore, changing C8 to 0.01uF along with the resistors will keep the time constants very similar - changing the button release time constant from 0.002 to 0.0027 and the button press time constant from 0.001 to 0.0013.*

OR

**Fix No. 2:**
- Change R5 to 49.9kΩ: https://mou.sr/3Q3NRZO
This is R10 on the multicart board. You may have extras.

*Technical explanation: Instead of the SRAM voltage dropping to 1/2 of the battery voltage when the button is pressed, it will drop to 83% of the battery voltage. This is still plenty of headroom for the SRAM to retain data. But keeping C8 the same will change the time constant for button release to 0.007 instead of 0.002.*

The differences between the two fixes are minor, but if I had to pick a "proper" one I would go with Fix No. 1. That is what I'll be implementing into the BOM going forward. The reason for this is that the time constants will remain nearly unchanged between the new part values and the old part values - with just changing R5, the RC time constant for the rising edge of the button press is slower. I don't believe it is slow enough to cause any issues, the only thing that could cause a problem is if the CPU starts up before the ROM bank switching is done, but the time constant for the CPU power-on reset is 0.016 at the fastest. Experimentally, both ways seem to work.

--- TL;DR ---

If you made an MBC3 or MBC5 multicart board before February 24, 2024:

- Change R5 and R6 to 130kΩ, and C8 to 0.01uF.

OR

- Change R5 to 49.9kΩ.

If you don't, under certain circumstances you may lose save data while pressing the button to change games while the Game Boy is powered off. **The proper part values have been updated in the BOM as of February 24, 2024.**
